### ISSUE TITLE
rye sync only works from the root

### ISSUE THREAD

**Issue by bluss** [mitsuhiko/rye#284]
### Steps to Reproduce

1. rye init nested
2. cd nested
3. mkdir test
4. cd test
5. rye sync

### Expected Result

rye sync could emit user friendly error or work when running in a nested directory

### Actual Result

```
pip._internal.exceptions.InstallationError: file:///. (from -r /tmp/.tmpBCmQ3R (line 1)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
Error: could not write production lockfile for project

Caused by:
    failed to generate lockfile
```

### Version Info

rye 0.5.0
commit: 0.5.0 (5f96294de 2023-05-31)
platform: linux (x86_64)
self-python: cpython@3.10
symlink support: true

**Comment by dsp** [comment]
I added a quick fix. It's not very well tested and I'll give it more thought and testing before asking @mitsuhiko to merge.

**Comment by bluss** [comment]
Thanks! Fixed by #292
